### PR TITLE
[Linux/X11] Add a default error handler for X11 to avoid crashes.

### PR DESCRIPTION
**Pull request by ghost**
The default behaviour for X11 is to crash even on non-fatal errors when there is no error handler set. This change allows the window to stay open and may enable users to save their work when things go wrong.

This acts as a workaround for #65425 and #68471

### CODE REVIEW

**Comment by mxnemu** [comment]
clang-format makes the multiline string unreadable. Any advice on that?

**Review by deralmas — APPROVED** [review]
I haven't tested this but it looks fine!

**Comment by mxnemu** [comment]
Since the bugs aren't super easy to reproduce without installing a new window manager and configuring smart-borders, here are screenshots of my tests on `Ubuntu 22.04` `xmonad 0.15`.


**This branch** - The godot window stays open and usable when an error occures:

2023-03-20-210206_784x294_scrot

**This branch** - Using `xkill` still ends godot, just the same as in the official release. So no unintended effects here.

2023-03-20-205415_769x736_scrot


**Godot 4.0 before the patch** -  the window crashes on focus change as described in the issue reports:

2023-03-20-204148_777x684_scrot

File: platform/linuxbsd/x11/display_server_x11.cpp [hunk 1]
```diff
@@ -821,12 +821,30 @@ Size2i DisplayServerX11::screen_get_size(int p_screen) const {
...
+// and is much more complex, so this is a less complete simplified error X11 printer.
+int default_window_error_handler(Display *display, XErrorEvent *error) {
+	static char message[1024];
+	XGetErrorText(display, error->error_code, message, sizeof(message));
+
+	ERR_PRINT(vformat("Unhandled XServer error: %s"
+					  "\n   Major opcode of failed request: %d"
+					  "\n   Serial number of failed request: %d"
+					  "\n   Current serial number in output stream: %d",
+			String(message), error->request_code, error->minor_code, error->serial));
```
Comment (bruvzg) [inline platform/linuxbsd/x11/display_server_x11.cpp:838]:
```suggestion
			String::utf8(message), error->request_code, error->minor_code, error->serial));
```

According to docs: "The returned text is in the encoding of the current locale.", so it's likely UTF-8 in most cases, `String(x)` is for ASCII only.

**Review by bruvzg — APPROVED** [review]

**Comment by akien-mga** [comment]
For the record, your commit seems not to be linked to your GitHub account. See: [Why are my commits linked to the wrong user?](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/troubleshooting-commits/why-are-my-commits-linked-to-the-wrong-user) for more info.

File: platform/linuxbsd/x11/display_server_x11.cpp (hunk 1 above)
Comment (akien-mga) [reply platform/linuxbsd/x11/display_server_x11.cpp:838]: @mxnemu Are you able to do this change? This seems to be the only step left before merging.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**